runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcMinimapOverlay.java

```diff
@@ -56,9 +56,12 @@
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
+		
 		for (NPC npc : plugin.getHighlightedNpcs())
 		{
-			renderNpcOverlay(graphics, npc, npc.getName(), config.getHighlightColor());
+			if ((int) Math.hypot(client.getLocalPlayer().getWorldLocation().getX() - npc.getWorldLocation().getX(), client.getLocalPlayer().getWorldLocation().getY() - npc.getWorldLocation().getY()) < config.drawDistance()){
```

Comment:
There is `WorldPoint.distanceTo` and also `LocalPoint.distanceTo`. One of them probably does what you want.